src/adam_core/orbits/variants.py

```diff
@@ -168,6 +168,57 @@ def collapse(self, orbits: Orbits) -> Orbits:
 
         return qv.concatenate(orbits_list)
 
+    def collapse_by_object_id(self) -> Orbits:
+        """
+        Collapse the variant orbits into a mean and covariance matrix.
+
+        Returns
+        -------
+        collapsed_orbits : `~adam_core.orbits.orbits.Orbits`
+            The collapsed orbits.
+        """
+        # Group the variants by object_id
+        unique_object_ids = self.object_id.unique()
+
+        orbits = Orbits.empty()
+        for object_id in unique_object_ids:
+            object_variants = self.select("object_id", object_id)
+
+            # All the variants must have the same epoch
+            assert len(object_variants.coordinates.time.unique()) == 1
+
+            # Calculate the mean
+            mean = np.average(
+                object_variants.coordinates.values,
+                axis=0,
+            )
+
+            # Calculate the covariance matrix
+            covariance = weighted_covariance(
+                mean,
+                object_variants.coordinates.values,
+                np.ones(len(object_variants), dtype=np.float64),
```

Comment:
I think weights should sum to 1 but since its a constant value, I'm not sure it will have an impact on the result.

```suggestion
                np.ones(len(object_variants), dtype=np.float64) / len(object_variants),
```